automata/fa/dfa.py

```diff
@@ -598,6 +598,79 @@ def isfinite(self):
         except nx.exception.NetworkXNoCycle:
             return True
 
+    @classmethod
+    def contains_subsequence(cls, subsequence, input_symbols):
+        """
+        Creates a DFA which accepts all words which contain a specific subsequence of symbols
+        """
+        transitions = {0: {symbol: 0 for symbol in input_symbols}}
+
+        for prev_state, symbol in enumerate(subsequence):
+            next_state = prev_state + 1
+            transitions[next_state] = {symbol: next_state for symbol in input_symbols}
+            transitions[prev_state][symbol] = next_state
+
+        return cls(
+            states=set(transitions.keys()),
+            input_symbols=input_symbols,
+            transitions=transitions,
+            initial_state=0,
+            final_states={len(subsequence)},
+        )
+
+    @classmethod
+    def of_length(cls, min_length, max_length, input_symbols):
+        """
+        Creates a DFA which accepts all words whose length is between `min_length` and `max_length`, inclusive.
+        To allow infinitely long words the value `float('inf')` can be passed in for `max_length`.
+        """
+        transitions = {}
+        length_range = range(min_length) if max_length == float('inf') else range(max_length+1)
+        for prev_state in length_range:
+            next_state = prev_state + 1
+            transitions[prev_state] = {symbol: next_state for symbol in input_symbols}
+        last_state = len(transitions)
+        transitions[last_state] = {symbol: last_state for symbol in input_symbols}
+        final_states = {last_state} if max_length == float('inf') else set(range(min_length, max_length+1))
+        return cls(
+            states=set(transitions.keys()),
+            input_symbols=input_symbols,
+            transitions=transitions,
+            initial_state=0,
+            final_states=final_states,
+        )
+
+    @classmethod
+    def universal_language(cls, input_symbols):
+        return cls.contains_subsequence('', input_symbols)
+
+    @classmethod
+    def empty_language(cls, input_symbols):
+        return ~cls.universal_language(input_symbols)
```

Comment:
Kindof a nitpick, but using `~` actually involves some unnecessary function call overhead (since this operation minimizes by default), I think it would be better just to hardcode this construction (since it's only a couple lines long and can be copied from elsewhere).

Reply:
Updated